src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -178,6 +178,18 @@ void Terminal::UpdateSettings(winrt::Microsoft::Terminal::Settings::ICoreSetting
         proposedTop -= (proposedBottom - bufferSize.Y);
     }
 
+    Cursor& cursor = _buffer->GetCursor();
+    auto cursorPosition = cursor.GetPosition();
+
+    // If the cursor is positioned beyond the range of the viewport, then
+    // set the cursor position to a legal value.
+    cursorPosition.X = std::min(cursorPosition.X, static_cast<SHORT>(viewportSize.X - 1));
+    cursorPosition.Y = std::min(cursorPosition.Y, static_cast<SHORT>(viewportSize.Y - 1));
+
+    cursor.StartDeferDrawing();

```

Comment:
This looks strange but it's necessary. Because `SetPosition` will call `RedrawCursor`, and eventually find its way to `Terminal::IsCursorDoubleWidth()`, causing the crash. So I had to disable redrawing first, correct the position, and enable redrawing.  IMO the `SetPosition` (and related method) are too aggresive doing redrawing, which is not a good design.

Reply:
Yeah, you're right. We should likely fix the design on the redrawing and come up with something better. We can take this for now. I know it's ugly and complicated to fix the right way, but I'm glad you found something workable.